src/plugins/visualizations/common/expression_functions/vis_dimension.ts

```diff
@@ -54,28 +54,35 @@ export const visDimension = (): ExpressionFunctionDefinition<
     },
     format: {
       types: ['string'],
-      default: 'string',
       help: i18n.translate('visualizations.function.visDimension.format.help', {
         defaultMessage: 'Format',
       }),
     },
     formatParams: {
       types: ['string'],
-      default: '"{}"',
       help: i18n.translate('visualizations.function.visDimension.formatParams.help', {
         defaultMessage: 'Format params',
       }),
     },
   },
   fn: (input, args) => {
     const accessor = findAccessorOrFail(args.accessor, input.columns);
+    const column = typeof accessor === 'number' ? input.columns[accessor] : accessor;
+    const columnFormat = column.meta.params;
+    // if user doesn't specify format and format for column is not specified
+    // that we should use default format id ('string')
+    const format =
+      args.format || args.formatParams || !columnFormat
```

Comment:
It seems to me, `columnFormat.id` and `columnFormat.params` are optional at the `SerializedFieldFormat`. So, to prevent wrong behavior, it seems to me, you need to check for them too (for at least one of them is present, either fall back to default values to be safe. WDYT?

Reply:
@VladLasitsa, about this comment, never mind. I think it was working like that before and we don't need to overthink this situation.